add dark mode

nikosalonen opened the pull request:
This PR adds ✨dark mode✨!

* Theme switcher in top right corner
* Checks what is the preferred browser theme

Other stuff:
* Add `Number.` to `parseInt` to use the modularized globals
* Error handling for API calls
* Update mobile layout a little

![image](https://github.com/user-attachments/assets/4dc7ab6e-e3f1-4e93-9588-9470c5188463)
![image](https://github.com/user-attachments/assets/87836b3e-4767-454b-bfca-1067aea8b7f6)

nikosalonen committed:
Add theme switching functionality and update styles

- Implemented theme toggle button in the UI for dark/light mode.
- Added theme switching logic in app.js to handle user preferences.
- Updated styles.css to support theme-based styling with CSS variables.
- Included SVG icons for dark and light modes.

nikosalonen committed:
Update parseInt calls to use Number.parseInt for consistency.

nikosalonen committed:
Improve event listener handling by removing old listeners before adding new ones.

nikosalonen committed:
add error handling for api calls

nikosalonen committed:
Refactor theme handling and improve code consistency

- Added a class to the dash icon for improved styling.
- Cleaned up SVG file by removing unnecessary attributes and ensuring proper formatting.

nikosalonen committed:
Enhance styling for h1 and fieldset elements in styles.css

- Added text alignment, margin, and font size to h1 for better presentation.
- Updated fieldset styles to improve layout and added z-index for better stacking context.

nikosalonen committed:
Enhance table styling for mobile responsiveness in styles.css

- Added responsive styles for tables, tbody, and td elements to improve layout on smaller screens.
- Introduced new styles for playoff matchup cells and adjusted logo display for better alignment.
- Improved visibility of correct/incorrect picks with background color changes.

nikosalonen committed:
Update span styles in styles.css for improved theming and visibility

- Changed span text color to use CSS variable for better theme support.
- Adjusted opacity for span and span.correct-games for enhanced visibility.
- Added dark theme styling for span elements to improve contrast.

nikosalonen committed:
Refactor app.js for improved code consistency and readability

- Replaced forEach with for...of loops for better performance and clarity.
- Changed variable declarations from let to const where applicable to enhance immutability.

nikosalonen committed:
Add logo URL generation function and theme change listener

- Introduced `getLogoUrl` function to streamline logo URL generation based on theme.
- Updated logo source assignments in `createHeaders` and `createRow` functions to use the new utility.
- Added a MutationObserver to listen for theme changes and trigger table re-rendering accordingly.

nikosalonen committed:
fix code formatting

Hamatti commented on app.js:
```diff
@@ -173,36 +201,36 @@ function createRow(entry, tr, games, teams) {
     // If user's pick is not in the running anymore
-    if (userPick != game.team_1_id && userPick != game.team_2_id) {
-      selectedPick.src = `dash.svg`;
+    if (userPick !== game.team_1_id && userPick !== game.team_2_id) {
```
The APIs are quite inconsistent as how they store different values and thus the `==` was intentional here as I was lazy to convert everything to a consistent one.

This change currently breaks the app as the `userPick` and `game.[team_id]` are not of the same type.

nikosalonen replied:
Yeah just noticed this myself also, will revert.

nikosalonen committed:
revert accidental != -> !==

nikosalonen committed:
Fix code formatting in createRow function for consistency

nikosalonen committed:
Update table cell styles in styles.css for improved layout and clarity

- Added gap between flex items in table cells.
- Removed margin-right from pseudo-element for better alignment.
- Introduced a new pseudo-element to display "Pick:" label with adjusted styling.

Hamatti reviewed (approved):
Great stuff, thanks for the PR!
